File: business/business-api/src/main/java/pl/przybylo/przychodnia/business/GabinetService.java
package pl.przybylo.przychodnia.business;

import pl.przybylo.przychodnia.dto.gabinet.GabinetEditDto;
import pl.przybylo.przychodnia.dto.gabinet.GabinetNewDto;
import pl.przybylo.przychodnia.dto.gabinet.GabinetViewDto;

import java.util.List;

public interface GabinetService {

    List<GabinetViewDto> getGabinetList();

    GabinetViewDto add(GabinetNewDto gabinetNewDto);

    GabinetViewDto update(GabinetEditDto gabinetEditDto);

}

File: web/src/main/app/project/src/app/service/gabinet.service.ts
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {Injectable} from "@angular/core";
import {GabinetEditDto, GabinetNewDto, GabinetViewDto} from "@przychodnia/model/backend-model";

@Injectable()
export class GabinetService {

    constructor(private httpClient: HttpClient) {
    }

    public getGabinetList(): Observable<Array<GabinetViewDto>> {
        return <Observable<Array<GabinetViewDto>>>this.httpClient.get('/gabinety');
    }

    public add(gabinetNewDto: GabinetNewDto): Observable<GabinetViewDto> {
        return <Observable<GabinetViewDto>>this.httpClient.post('/gabinety', gabinetNewDto);
    }

    public update(gabinetViewDto: GabinetEditDto): Observable<GabinetViewDto> {
        return <Observable<GabinetViewDto>>this.httpClient.post('/gabinety', gabinetViewDto);
    }l

}
